### PR TITLE
Kubevirt switch to instancetype

### LINKED ISSUE #1452: KubeVirt switch to instancetype
As a user I want to swtich from presets to instancetype/preferences.

### PR DESCRIPTION
**Pull request by hdurand0710**
Signed-off-by: Helene Durand <helene@kubermatic.com>

**What this PR does / why we need it**:
KubeVirt support of instancetype and preferences  and deprecation of flavor.
It also includes adding some unit tests.
The deprecation of `flavor` will happen in 2 steps:
1) this PR: flavor and instancetype are still in use, no field is deprecated yet
2) integration and use of instancetype in KKP
3) dashboard update
4) new machine-controller and KKP with deprecation of `flavor`

This will allow to keep `dev` working (including dashboard) until the migration is over.

**Which issue(s) this PR fixes**:

Fixes #1452 

**What type of PR is this?**

/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
KubeVirt: support of instancetype and preferences (deprecation of flavor).
```

**Documentation**:

```documentation
TBD (will be done when last PR is merged in the Epic and UI is designed).
```

### CODE REVIEW

File: pkg/cloudprovider/provider/kubevirt/provider.go [hunk 1]
```diff
@@ -445,8 +455,14 @@ func (p *provider) Validate(ctx context.Context, spec clusterv1alpha1.MachineSpe
 	if err != nil {
 		return fmt.Errorf("failed to parse config: %w", err)
 	}
-	// If VMIPreset is specified, skip CPU and Memory validation.
-	if c.FlavorName == "" {
+	// If instancetype is specified, skip CPU and Memory validation.
+	// Values will come from instancetype.
+	if c.Instancetype == nil {
```
Comment (sankalp-r) [inline pkg/cloudprovider/provider/kubevirt/provider.go:460]: What if we combine:
```
if c.Instancetype == nil && c.FlavorName == "" {
...
}
```
As inside both conditions, it is doing the same `pasreResource`?

File: examples/kubevirt-machinedeployment.yaml [hunk 2]
```diff
@@ -32,6 +32,15 @@ spec:
                 # If instead specified directly, this value should be a base64 encoded kubeconfig.
                 value: "<< KUBECONFIG_BASE64 >>"
             virtualMachine:
+              instancetype:
+                name: "standard-2"
+                kind: "VirtualMachineInstancetype" # Allowed values: "VirtualMachineInstancetype"/"VirtualMachineClusterInstancetype"
+              preference:
+                name: "sockets-advantage"
+                category: "VirtualMachinePreference" # Allowed values: "VirtualMachinePreference"/"VirtualMachineClusterPreference"
```
[the remark this answers is not in the document]
Comment (hdurand0710) [reply examples/kubevirt-machinedeployment.yaml:40]: oops, thanks, done in the same next PR as the switch to 1.55.0.

**Pull request merged**